### PR TITLE
Gurucul Risk Analytics

### PR DESCRIPTION
**Pull request by graadmin**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_test.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
+import json
+import io
+
+
+def util_load_json(path):
+    with io.open(path, mode='r', encoding='utf-8') as f:
+        return json.loads(f.read())
+
+
```
Comment (DeanArbel) [inline Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_test.py:9]: Unit tests should contain given-when-then comment to describe the unit test scenario. Given <some input>, When <some state> Then <ensure some behavior>.
This comment applies to all unit tests below. Please make sure to go over all of them and add the missing details.

File: Packs/Gurucul/Playbooks/playbook-GraCase.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
+id: GraCase
+version: 1
```
Comment (DeanArbel) [inline Packs/Gurucul/Playbooks/playbook-GraCase.yml:2]:
```suggestion
version: -1
```

File: Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
+id: 3428794d-0afe-4810-8b3b-038a499ef8bd
+version: 1
```
Comment (DeanArbel) [inline Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml:2]:
```suggestion
id: GraHighRiskUser
version: -1
```

**Review by DeanArbel — CHANGES_REQUESTED** [review]
Very nice job!
* The integration logo doesn't match the requirements (120x50 px PNG, transparent background, <10kB in Size) - the size you provided is 230x53.
* See the rest of my comments below.

File: Packs/Gurucul/pack_metadata.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+    "name": "Gurucul Risk Analytics",
+    "description": "Gurucul Risk Analytics (GRA) is a Unified Security and Risk Analytics platform.",
+    "support": "community",
```
Comment (DeanArbel) [inline Packs/Gurucul/pack_metadata.json:4]:
```suggestion
    "support": "partner",
```

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+            if incident_created_time > latest_created_time:
+                latest_created_time = int(incident_created_time)
+
+        next_run = {'last_fetch': latest_created_time}
+    return next_run, incidents
+
+
+def test_module_command(client: Client) -> str:
+    try:
+        client.validate_api_key()
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.py:193]: Resolved, Check-in next commit

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 6]
```diff
@@ -0,0 +1,761 @@
...
+description: This is the GRA integration for getting started.
+configuration:
+- display: Server URL (e.g. https://example.net)
+  name: url
+  defaultvalue: https://example.net
+  type: 0
+  required: true
+- display: Authorization Key
+  name: apikey
+  type: 0
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml]: Resolved, Check-in next commit

File: Packs/Gurucul/Classifier/classifier-GraCase_5_5_0.json [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+			"accessor": "",
+			"filters": [],
+			"root": "incidentType",
+			"transformers": []
+		},
+		"simple": ""
+	},
+	"unclassifiedCases": {},
+	"version": -1
+}
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-GraCase_5_5_0.json]: Resolved, Check-in next commit

File: Packs/Gurucul/Classifier/classifier-Gurucul-GRACase.json [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+						"root": "entity",
+						"transformers": []
+					},
+					"simple": ""
+				}
+			}
+		}
+	},
+	"name": "Gurucul-GRACases (custom) (classifier)",
+	"nameRaw": "Gurucul-GRACases (custom) (classifier)",
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-Gurucul-GRACase.json]: Resolved, Check-in next commit

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 9]
```diff
@@ -0,0 +1,761 @@
+commonfields:
+  id: Gurucul-GRA
+  version: -1
+name: Gurucul-GRA
+display: Gurucul-GRA
+category: Analytics & SIEM
+description: This is the GRA integration for getting started.
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml]: Resolved, Check-in next commit

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_description.md [hunk 10]
```diff
@@ -0,0 +1,3 @@
+## Hello World
+- This text is markdown
+- Here you should explain how to configure the instance in Demisto
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_description.md]: Resolved, Check-in next commit

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_test.py (hunk 1 above)
Comment (graadmin) [reply Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_test.py:9]: Resolved, Check-in next commit

File: Packs/Gurucul/Playbooks/playbook-GraCase.yml (hunk 2 above)
Comment (graadmin) [reply Packs/Gurucul/Playbooks/playbook-GraCase.yml]: Resolved, Check-in next commit

File: Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml (hunk 3 above)
Comment (graadmin) [reply Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml]: Resolved, Check-in next commit

File: Packs/Gurucul/pack_metadata.json (hunk 4 above)
Comment (graadmin) [reply Packs/Gurucul/pack_metadata.json]: Resolved, Check-in next commit

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_description.md [hunk 11]
```diff
@@ -1,3 +1,2 @@
-## Hello World
-- This text is markdown
-- Here you should explain how to configure the instance in Demisto
\ No newline at end of file
+## Gurucul GRA
+- Please contact Gurucul for getting APIKEY
```
Comment (DeanArbel) [inline Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA_description.md:2]: Is there a suggested method of contact? e.g. support email.

File: Packs/Gurucul/ReleaseNotes/1_1_0.md [hunk 12]
```diff
@@ -1,45 +0,0 @@
...
-|gra-cases|Retrieve list of all Cases for a given Status|status|
-|gra-user-anomalies|Retrieve Behavioral Anomalies triggered for a given User|employee_id|
-
->##Playbooks
-> Below are then 2 playbooks
->
-    1.GraHighRiskUsers
-    2.GraUsersCases
-    
-
```
Comment (DeanArbel) [inline Packs/Gurucul/ReleaseNotes/1_1_0.md]: Was this removed by accident?

**Review by DeanArbel — COMMENTED** [review]
You have a few fields that should be removed from your files (see the validation errors for the full list) - these are not part of the scheme and can be safely removed.
Let's do a review next week. I'll contact you by email to schedule a time.

**Comment by graadmin** [comment]
Hi,

As per the suggestion of build suggestion.
Removed from repository

Thanks & Regards



Umesh Balshetwar

Technical Lead

*Desk Ext*: 36

*Mobile*: 91- 9975885832

*Skype*: Umesh.Balshetwar

Gurucul

File: Packs/Gurucul/Classifier/classifier-mapper-Gurucul-HighRiskUsers.json [hunk 13]
```diff
@@ -0,0 +1,46 @@
+{
+	"brands": [
+		"Gurucul-GRA"
+	],
```
Comment (DeanArbel) [inline Packs/Gurucul/Classifier/classifier-mapper-Gurucul-HighRiskUsers.json:4]: This field should be removed.

```suggestion
```

**Review by DeanArbel — COMMENTED** [review]
I have compiled for you below the changes that are required by the validation.

File: Packs/Gurucul/ReleaseNotes/1_1_0.md (hunk 12 above)
Comment (graadmin) [reply Packs/Gurucul/ReleaseNotes/1_1_0.md]: suggestion as per circleCI app , removed
but as per your suggestion again I will add to repository

File: Packs/Gurucul/Playbooks/playbook-GraCase.yml [hunk 14]
```diff
@@ -0,0 +1,140 @@
+id: GraCase
+version: -1
+fromversion: 5.0.0
+vcShouldKeepItemLegacyProdMachine: false
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Playbooks/playbook-GraCase.yml]: Removed

File: Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml [hunk 15]
```diff
@@ -0,0 +1,140 @@
+id: 3428794d-0afe-4810-8b3b-038a499ef8bd
+version: -1
+fromversion: 5.0.0
+vcShouldKeepItemLegacyProdMachine: false
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Playbooks/playbook-GraHighRiskUser.yml]: Removed

File: Packs/Gurucul/Classifier/classifier-Gurucul-GRACase.json [hunk 16]
```diff
@@ -0,0 +1,38 @@
+{
+	"brands": [
+		"Gurucul-GRA"
+	],
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-Gurucul-GRACase.json]: Removed

File: Packs/Gurucul/Classifier/classifier-Gurucul-HighRiskUsers.json [hunk 17]
```diff
@@ -0,0 +1,39 @@
+{
+	"brands": [
+		"Gurucul-GRA"
+	],
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-Gurucul-HighRiskUsers.json]: Removed

File: Packs/Gurucul/Classifier/classifier-mapper-Gurucul-GRACases.json [hunk 18]
```diff
@@ -0,0 +1,127 @@
+{
+	"brands": [
+		"Gurucul-GRA"
+	],
```
[the remark this answers is not in the document]
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-mapper-Gurucul-GRACases.json]: Removed

File: Packs/Gurucul/Classifier/classifier-mapper-Gurucul-HighRiskUsers.json (hunk 13 above)
Comment (graadmin) [reply Packs/Gurucul/Classifier/classifier-mapper-Gurucul-HighRiskUsers.json]: Removed

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 19]
```diff
@@ -0,0 +1,762 @@
...
+      description: Event Desc
+      type: string
+    - contextPath: Gra.User.Activity.event_date
+      description: Event Date
+      type: date
+    - contextPath: Gra.User.Activity.risk_score
+      description: Risk Score
+      type: number
+    description: Retrieve User Activity for a Given User
+  - name: fetch-gra-users-details
```
Comment (DeanArbel) [inline Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml:581]: Please change the name of this command to `gra-fetch-users-details`

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 20]
```diff
@@ -0,0 +1,762 @@
...
+    - contextPath: Gra.User.title
+      description: Title
+      type: string
+    - contextPath: Gra.User.joingDate
+      description: Joining Date
+      type: date
+    - contextPath: Gra.User.exitDate
+      description: Exit Date
+      type: date
+    description: get details of the user
```
Comment (DeanArbel) [inline Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml:629]:
```suggestion
    description: Get details of the user
```

**Review by DeanArbel — COMMENTED** [review]
Please go over my comments.
Currently the build is not passing, so some changes are required.

**Comment by DeanArbel** [comment]
Following our demo, please add `Account` standard output (and dbot score) where applicable.

See:
### YML Output standards:
https://xsoar.pan.dev/docs/integrations/context-standards-mandatory#email-object
https://xsoar.pan.dev/docs/integrations/context-standards-mandatory#dbot-score

### PY Output using CommandResults
https://github.com/demisto/content/blob/master/Packs/EmailRepIO/Integrations/EmailRepIO/EmailRepIO.py#L113

File: Packs/Gurucul/pack_metadata.json [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Gurucul Risk Analytics (GRA) is a Unified Security and Risk Analytics platform.",
+    "support": "partner",
+    "currentVersion": "1.0.0",
+    "author": "Gurucul",
+    "url": "https://www.gurucul.com",
+    "email": "support@gurucul.com",
+    "githubUser": "graadmin",
+    "created": "2020-12-02T18:00:00Z",
+    "categories": [
+        "Analytics & SIEM"
```
Comment (DeanArbel) [inline Packs/Gurucul/pack_metadata.json:12]: I believe the type of the pack/integration should be `Data Enrichment & Threat Intelligence`.
See: https://xsoar.pan.dev/docs/concepts/use-cases#data-enrichment--threat-intelligence
```suggestion
        "Data Enrichment & Threat Intelligence"
```

File: Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml [hunk 22]
```diff
@@ -0,0 +1,762 @@
+commonfields:
+  id: Gurucul-GRA
+  version: -1
+name: Gurucul-GRA
+display: Gurucul-GRA (Community Contribution)
+category: Analytics & SIEM
```
Comment (DeanArbel) [inline Packs/Gurucul/Integrations/GuruculGRA/GuruculGRA.yml:6]: I believe the type of the pack/integration should be `Data Enrichment & Threat Intelligence`.
See: https://xsoar.pan.dev/docs/concepts/use-cases#data-enrichment--threat-intelligence
```suggestion
category: Data Enrichment & Threat Intelligence
```

**Review by DeanArbel — APPROVED** [review]
@graadmin Per our conversation, we're ok that the support for DBotScore for `Email` will be postponed to your next phase.

Please note @upbalshetwar also needs to sign the CLA before this pull request can be merged because he's also contributed to this pack.

I have 1 final comment, please go over it and change the Pack type accordingly (unless you believe it should be `Analytics & SIEM` - in that case just comment below saying that).

**Pull request merged**